core/src/main/java/org/apache/iceberg/rest/RESTUtil.java

```diff
@@ -215,4 +215,19 @@ public static Namespace decodeNamespace(String encodedNs) {
 
     return Namespace.of(levels);
   }
+
+  public static String buildCredentialEndpoint(String catalogUri, String credentialEndpoint) {
```

Comment:
Can we make this more generic?  Basically we're saying that we need a utility to either resolve a full qualified uri or a relative one, which I don't feel needs to be credential specific.

Reply:
good point, I've renamed this to `resolveEndpoint`